File: 611. Valid Triangle Number/Python/Valid Triangle Number.py
class Solution:
    def triangleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        nums.sort()
        for i in range(2, len(nums)):
            first = 0
            second = i - 1
            while first < second:
                if nums[first] + nums[second] > nums[i]:
                    result += second - first
                    second -= 1
                else:
                    first += 1
        return result

File: 611. Valid Triangle Number/C++/Valid Triangle Number.cpp
class Solution {
public:
    int triangleNumber(vector<int> &nums) {
        int result = 0;
        sort(nums.begin(), nums.end());
        for (int i = 2; i < nums.size(); i++) {
            int first = 0;
            int second = i - 1;
            while (first < second) {
                if (nums[first] + nums[second] > nums[i]) {
                    result += second - first;
                    second--;
                } else {
                    first++;
                }
            }
        }
        return result;
    }
};